app/Filament/Resources/NodeResource/RelationManagers/AllocationsRelationManager.php

```diff
@@ -122,8 +132,19 @@ public function table(Table $table): Table
 
                                     $start = max((int) $start, 0);
                                     $end = min((int) $end, 2 ** 16 - 1);
-                                    foreach (range($start, $end) as $i) {
-                                        $ports->push($i);
+                                    $range = $start <= $end ? range($start, $end) : range($end, $start);
+                                    foreach ($range as $i) {
+                                        if ($i > 1024 && $i <= 65535) {
+                                            if (Allocation::query()->where('ip', $get('allocation_ip'))->where('port', $portEntry)->exists()) {
```

Comment:
This needs to happen after the foreach loop because of the [N+1 problem](https://stackoverflow.com/questions/97197/what-is-the-n1-selects-problem-in-orm-object-relational-mapping)!